src/core/pointcloud/qgspointcloudlayerexporter.h

```diff
@@ -45,6 +45,51 @@ class CORE_EXPORT QgsPointCloudLayerExporter SIP_NODEFAULTCTORS
 {
   public:
 
+    /**
+     * Supported export formats for point clouds
+     */
+    enum class ExportFormat : int
+    {
+      Memory = 0, //!< Memory layer
+      Las = 1, //!< LAS/LAZ point cloud
+      Gpkg = 2, //!< Geopackage
+      Shp = 3, //!< ESRI ShapeFile
+      Dxf = 4, //!< AutoCAD dxf
+    };
+
+    /**
+     * Gets a list of the supported export formats.
+     *
+     * \see setFormat()
+     */
+    static QList< ExportFormat > supportedFormats() SIP_SKIP
+    {
+      QList< ExportFormat > formats;
+      formats << ExportFormat::Memory;
+#ifdef HAVE_PDAL_QGIS
+      formats << ExportFormat::Las;
+#endif
+      formats << ExportFormat::Gpkg;
+      formats << ExportFormat::Shp;
+      formats << ExportFormat::Dxf;
+      return formats;
+    }
+
+    /**
+     * Gets the translated name for the specified \a format
+     */
+    static QString getTranslatedName( ExportFormat format );
+
+    /**
+     * Gets the extensions filter for the specified \a format
+     */
+    static QString getFilter( ExportFormat format );
+
+    /**
+     * Gets the OGR driver name for the specified \a format
+     */
+    static QString getOgrDriverName( ExportFormat format );
```

Comment:
I would move this one out of the public API - it feels like implementation detail...

Reply:
SIP_SKIPped!